### PR TITLE
[Issue-1219] - when generating dnssec keys for a cdn, figure out a cdn's delivery services by using cdn_id rather than prof…

### LINKED ISSUE #1219: Generating DNSSEC keys for a brand new CDN thru the API creates DNSSEC keys for potentially the wrong delivery services and then results in a 500 internal server error
The following API results in a 500 ISE when you try to generate DNSSEC keys for a brand-spanking new CDN: 

$r->post("/api/$version/cdns/dnsseckeys/generate

Here is the error:

[2017-09-21 15:50:00,413] [ERROR] DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR:  null value in column "profile" violates not-null constraint
DETAIL:  Failing row contains (null, 3399, 2017-09-21 15:50:00.391382-06). [for Statement "INSERT INTO profile_parameter ( parameter, profile) VALUES ( ?, ? ) RETURNING profile" with ParamValues: 1='3399', 2=undef] at /code/src/github.com/apache/incubator-trafficcontrol/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm line 165

This is because on this line:

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm#L91

If $profile_id is undef (which it will be because I have setup no traffic router servers for this CDN yet and therefore there is no profile to fetch), it will cause the following problems:

1. DNSSEC keys are created for all delivery services that have no profile regardless of CDN because of the result of this query when profile = undef

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm#L95

2. The error stated above because you can't create a profileparameter record with profile = undef

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm#L165

TLDR; 

There is a problem when profile can't be found on this line:

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm#L91

1. why is profile being used to decide which delivery services to create dnssec keys for? 

https://github.com/apache/incubator-trafficcontrol/blob/master/traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm#L94

Each DS has a CDN so why not figure out the DS's like this

my @ds_rs = $self->db->resultset('Deliveryservice')->search( { cdn_id => cdn.id } , { prefetch => [ { 'cdn' => undef }]});

2. Can we skip the profile parameter part if no profile was ever found?

### PR DESCRIPTION
**Pull request by mitchell852**
…ile; also no need to update an 'inception' parameter

fixes #1219

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/309/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/268/

File: traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm [hunk 1]
```diff
@@ -86,12 +86,9 @@ sub register {
...
-			#delivery services
-			#first get profile_id
-			my $profile_id = $self->get_profile_id_by_cdn($key);
-
-			#then get deliveryservices
-			my %search = ( profile => $profile_id );
+			#find the cdn's delivery services to generate keys for
+			my $cdn = $self->db->resultset('Cdn')->find( { name => $key } );
+			my %search = ( cdn_id => $cdn->id );
 			my @ds_rs = $self->db->resultset('Deliveryservice')->search( \%search , { prefetch => [ { 'cdn' => undef }]});
```
Comment (dneuman64) [inline traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm]: couldn't you just define the CDN in the prefetch?  Right now it is undef

Comment (mitchell852) [reply traffic_ops/app/lib/MojoPlugins/DnssecKeys.pm]: new commit coming

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/332/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/291/

**Pull request merged**